specification/servicefabric/resource-manager/Microsoft.ServiceFabric/stable/2020-03-01/cluster.json

```diff
@@ -0,0 +1,1538 @@
+        },
+        "healthCheckStableDuration": {
+          "type": "string",
+          "description": "The amount of time that the application or cluster must remain healthy before the upgrade proceeds to the next upgrade domain. The duration can be in either hh:mm:ss or in d.hh:mm:ss.ms format."
+        },
+        "healthCheckRetryTimeout": {
+          "type": "string",
+          "description": "The amount of time to retry health evaluation when the application or cluster is unhealthy before the upgrade rolls back. The timeout can be in either hh:mm:ss or in d.hh:mm:ss.ms format."
+        },
+        "upgradeTimeout": {
+          "type": "string",
+          "description": "The amount of time the overall upgrade has to complete before the upgrade rolls back. The timeout can be in either hh:mm:ss or in d.hh:mm:ss.ms format."
+        },
+        "upgradeDomainTimeout": {
+          "type": "string",
+          "description": "The amount of time each upgrade domain has to complete before the upgrade rolls back. The timeout can be in either hh:mm:ss or in d.hh:mm:ss.ms format."
+        },
+        "healthPolicy": {
+          "$ref": "#/definitions/ClusterHealthPolicy",
+          "description": "The cluster health policy used when upgrading the cluster."
+        },
+        "deltaHealthPolicy": {
+          "$ref": "#/definitions/ClusterUpgradeDeltaHealthPolicy",
+          "description": "The cluster delta health policy used when upgrading the cluster."
+        }
+      },
+      "description": "Describes the policy used when upgrading the cluster."
+    },
+    "ClusterVersionDetails": {
+      "properties": {
+        "codeVersion": {
+          "type": "string",
+          "description": "The Service Fabric runtime version of the cluster."
+        },
+        "supportExpiryUtc": {
+          "type": "string",
+          "description": "The date of expiry of support of the version."
+        },
+        "environment": {
+          "$ref": "#/definitions/ClusterEnvironment",
+          "description": "Indicates if this version is for Windows or Linux operating system."
+        }
+      },
+      "description": "The detail of the Service Fabric runtime version result"
+    },
+    "DiagnosticsStorageAccountConfig": {
+      "required": [
+        "blobEndpoint",
+        "protectedAccountKeyName",
+        "queueEndpoint",
+        "storageAccountName",
+        "tableEndpoint"
+      ],
+      "properties": {
+        "storageAccountName": {
+          "type": "string",
+          "description": "The Azure storage account name."
+        },
+        "protectedAccountKeyName": {
+          "type": "string",
+          "description": "The protected diagnostics storage key name."
+        },
+        "protectedAccountKeyName2": {
+          "type": "string",
+          "description": "The secondary protected diagnostics storage key name."
+        },
```

Comment:
either keyNames should become an array or this should be "secondaryProtectedAccountKeyName".  Description should also state when this key would be used over the primary

Reply:
@pilor, I added more information to the description. Can we keep this name, though? we introduced this on version 2019-06-01-preview on this pr: https://github.com/Azure/azure-rest-api-specs/pull/7092 which was approved by ARM, If we make the change it will be a breaking change with that version. Also, can you also review 2019-11-01-preview https://github.com/Azure/azure-rest-api-specs/pull/8745? which has the same changes as this stable version but we are going to release the preview before